runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java

```diff
@@ -277,7 +280,11 @@ private void updateWarning(boolean weaponSwitch)
 				{
 					if (weaponSwitch)
 					{ // NOPMD EmptyIfStmt
-						// TODO : chat message to warn players that their weapon switch also caused an unwanted attack style change
+						final String msg = new ChatMessageBuilder()
+								.append(ChatColorType.HIGHLIGHT)
+								.append("Switching weapons changed your attack style to an unwanted style!")
```

Comment:
There are ways to have a different attack style selected via means other than switching weapons (running out of runes for autocasting, entering areas which automatically equip items such as  the gauntlet, etc.), so I'd prefer wording which doesn't necessarily reference that. Also, we know what weapon style is unwanted, so we should print the style in this message. imo this should be formatted:

> Your weapon is using a warned attack style: {STYLE}

Reply:
I've been having some trouble implementing this one. Whenever I switch to a staff that has defensive casting selected it seems to change the attack style to casting then defensive casting. Then when I switch to some other attack style such as Accurate with the staff equipped it switches to Accurate then defensive then Accurate again. I'm not sure what's causing these intermediate changes in the attack style. I get similar results when using the monitoring the VarPlayer.ATTACK_STYLE variable.

Reply:
It's possible then that you'll need to set a flag to check this on the next game tick (after all var changes have been processed) rather than alerting upon var change

Reply:
Would it be bad design to check this flag every game tick?

Reply:
It will effectively be the same either way. Either way you'll end up running:

```java
@Subscribe
void onGameTick(GameTick e)
{
	if (combatStyleVarChanged)
	{
		// ...
	}
}
```

or you'll have some field storing the var and comparing the previous to current attack styles, like what you're doing now. Both are effectively equally efficient and work the same way, so that should be fine.